app/src/main/java/com/arctouch/codechallenge/home/HomeAdapter.kt

```diff
@@ -9,21 +9,26 @@ import com.arctouch.codechallenge.model.Movie
 import com.arctouch.codechallenge.util.MovieImageUrlBuilder
 import com.bumptech.glide.Glide
 import com.bumptech.glide.request.RequestOptions
+import kotlinx.android.synthetic.main.layout_stars.view.*
 import kotlinx.android.synthetic.main.movie_item.view.*
 
-class HomeAdapter(private val movies: List<Movie>) : RecyclerView.Adapter<HomeAdapter.ViewHolder>() {
+class HomeAdapter(
+    private val movies: List<Movie>,
+    private val itemListener: MovieClickListener
+) : RecyclerView.Adapter<HomeAdapter.ViewHolder>() {
 
-    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
-
-        private val movieImageUrlBuilder = MovieImageUrlBuilder()
+    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
 
         fun bind(movie: Movie) {
             itemView.titleTextView.text = movie.title
             itemView.genresTextView.text = movie.genres?.joinToString(separator = ", ") { it.name }
-            itemView.releaseDateTextView.text = movie.releaseDate
+            itemView.releaseDateTextView.text = movie.releaseDate?.split("-")?.get(0)
+            itemView.ratingBar.rating = movie.rate/2
+            itemView.tvRate.text = itemView.context.getString(R.string.movie_rate, movie.rate)
+            itemView.setOnClickListener { itemListener.onMovieClick(movie) }
 
             Glide.with(itemView)
```

Comment:
We have the same code doing basically the same thing into ```DetailActivity```. How can we avoid the duplicate code on this situation?

Reply:
I created an util function to show the images, and also to set rating bar ([#26024585](https://github.com/diogojs/tmdb_app/pull/1/commits/26024585520cc150a6cd93b6b83662273e42bd19))